File: frontend/src/api/modules/poetFeatured.ts
import request from '@/utils/request'

export interface PoetFeatured {
  id: number
  poetId: number | null
  name: string
  dynasty: string
  description: string
  biography: string
  imageUrl: string
  sortOrder: number
  status: number
}

export function getRandomPoetFeatured(count: number = 4) {
  return request.get<PoetFeatured[]>('/poet-featured/random', { params: { count } })
}

export function getPoetFeaturedList() {
  return request.get<PoetFeatured[]>('/poet-featured/list')
}

export function getPoetFeaturedDetail(id: number) {
  return request.get<PoetFeatured>(`/poet-featured/${id}`)
}

File: sc-moyuan-backend/src/main/java/com/moyuan/entity/PoetFeatured.java
package com.moyuan.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;
import java.time.LocalDateTime;

@Data
@TableName("poet_featured")
public class PoetFeatured {
    @TableId(type = IdType.AUTO)
    private Long id;
    private Long poetId;
    private String name;
    private String dynasty;
    private String description;
    private String biography;
    private String imageUrl;
    private Integer sortOrder;
    private Integer status;
    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createTime;
    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updateTime;
    @TableLogic
    private Integer deleted;
}